### DISCUSSION TITLE
How do I work without LocalStorage?

### DISCUSSION THREAD

**Discussion by martinszeltins** [nuxt/nuxt#22915] [Questions] [answered]
I am trying to figure out how to migrate my Vue.js app to Nuxt.js with SSR support. In my Vue.js app I have this page where I load products. The user is a able to apply some filtering and sorting and then those filters get sent to the backend api and the backend (Laravel) returns the filtered products. It also has a feature where it saves all the filters that the user has selected in localStorage so that on next page load when the products are loaded from the backend API, it reads all the filters from localStorage and attaches them to the backend API request. As a result when the user opens the products page and if there were some filters saved in localStorage, the products page will open and all the filters will be applied.

The filters are stored in localStorage as stringified object like this and there could be many many filters:

[{"key":"F21","value":["2","6"],"comparisonOperator":"RANGE","location":"ToolSearchResults"},{"key":"A51","value":["0"],"comparisonOperator":"IN","location":"ToolSearchResults"}]

Now that I am migrating my app to Nuxt I have run into this problem where it says that the window object or localStorage are not defined when the page loads initially and the API call is being made and it is trying to read from localStorage to see if there are any filters saved.

It looks to me like I can't do with Nuxt what I was able to do with my SPA Vue app. What can I do now?

**Comment by logotip4ik** [comment]
You can wrap your calls to the window or localStorage into the `import.meta.client` block. This way everything inside will run only on the client side. Little example:

```js
if (import.meta.client) {
  filters.value = localStorage.getItem('filters')
}
```

**Comment by martinszeltins** [comment]
> You can wrap your calls to the window or localStorage into the `import.meta.client` block. This way everything inside will run only on the client side. Little example:
> 
> ```js
> if (import.meta.client) {
>   filters.value = localStorage.getItem('filters')
> }
> ```

But I want to access LocalStorage in SSR server-side. I want to make a request and in order to make the request I need to first read some data from LocalStorage. Client-side only solution won't work for me.

**Reply by logotip4ik** [reply, marked as answer]
Then you could try using cookies rather than localStorage. Nuxt provides [useCookie](https://nuxt.com/docs/api/composables/use-cookie) compostable to automatically serialize, deserialize data and it will make your filter options available on server side. Another option would be storing filtering data in query
